disnake/ui/action_row.py

```diff
@@ -0,0 +1,251 @@
+
+        self._underlying = ActionRowComponent._raw_construct(
+            type=ComponentType.action_row,
+            children=components,
+        )
+
+    @property
+    def children(self) -> List[Component]:
+        """List[:class:`Component`]: The components of this row."""
+        return self._underlying.children
+
+    @children.setter
+    def children(self, value: List[Component]):
+        self._underlying.children = value
+
+    @property
+    def type(self) -> ComponentType:
+        return self._underlying.type
+
+    def append_item(self, item: Item) -> None:
+        """Appends an item to the action row.
+
+        Parameters
+        -----------
+        item: :class:`disnake.ui.Item`
+            The item to append to the action row.
+
+        Raises
+        -------
+        ValueError
+            The width of the action row exceeds 5.
+        """
+        if self._total_width + item.width > 5:
+            raise ValueError("Too many items in this row, can not appen a new one.")
+
+        self._total_width += item.width
+        self._underlying.children.append(item._underlying)  # type: ignore
+
+    def add_button(
+        self,
+        *,
+        style: ButtonStyle = ButtonStyle.secondary,
+        label: Optional[str] = None,
+        disabled: bool = False,
+        custom_id: Optional[str] = None,
+        url: Optional[str] = None,
+        emoji: Optional[Union[str, Emoji, PartialEmoji]] = None,
+    ):
+        """Adds a button to the action row.
+
+        To append a pre-existing :class:`disnake.ui.Button` use the
+        :meth:`append_item` method instead.
+
+        Parameters
+        -----------
+        style: :class:`disnake.ButtonStyle`
+            The style of the button.
+        custom_id: Optional[:class:`str`]
+            The ID of the button that gets received during an interaction.
+            If this button is for a URL, it does not have a custom ID.
+        url: Optional[:class:`str`]
+            The URL this button sends you to.
+        disabled: :class:`bool`
+            Whether the button is disabled or not.
+        label: Optional[:class:`str`]
+            The label of the button, if any.
+        emoji: Optional[Union[:class:`.PartialEmoji`, :class:`.Emoji`, :class:`str`]]
+            The emoji of the button, if available.
+
+        Raises
+        -------
+        ValueError
+            The width of the action row exceeds 5.
+        """
+        if self._total_width >= 5:
+            raise ValueError("Too many items in this row, can not add a button.")
+
+        self._total_width += 1
+        self._underlying.children.append(
+            ButtonComponent._raw_construct(
+                type=ComponentType.button,
+                custom_id=custom_id,
+                url=url,
+                disabled=disabled,
+                label=label,
+                style=style,
+                emoji=emoji,
+            )
+        )
```

Comment:
Maybe also call self.append_item here? Would have to construct a `ui.Button` though, so not sure.

Also:
  - what about automatically defaulting to random `custom_id`s? (this would be taken care of by `ui.Button`, if we use it)
  - `ButtonComponent.emoji` is `PartialEmoji | None`, it can't take an `Emoji` instance
    (not sure if there are other type mismatches, this is just what I noticed pretty much by accident lol)